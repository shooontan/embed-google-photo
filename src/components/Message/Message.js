// @flow
import * as React from 'react';
import styled from 'styled-components';

type Props = {
  message: string,
};

const Message = ({ message }: Props) => {
  if (!message) {
    return null;
  }

  return (
    <StyledMessage>
      <p>{message}</p>
    </StyledMessage>
  );
};

const StyledMessage = styled.div`
  width: 100%;
  margin-top: -1em;
  text-align: center;

  & p {
    display: inline-block;
    margin: 0;
    padding: 1em;
    background: #333;
    color: #fafafa;
    border-radius: 2px;
    overflow: hidden;
  }
`;

export default Message;
