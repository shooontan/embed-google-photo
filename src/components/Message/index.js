// @flow
import Message from './Message';

export default Message;
