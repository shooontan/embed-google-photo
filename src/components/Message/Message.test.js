// @flow
import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import Message from './Message';

test('Form Component', () => {
  let testRenderer = TestRenderer.create(<Message message="" />);
  expect(testRenderer).toMatchSnapshot();

  testRenderer = TestRenderer.create(<Message message="this is Message Component Text" />);
  expect(testRenderer).toMatchSnapshot();
});
