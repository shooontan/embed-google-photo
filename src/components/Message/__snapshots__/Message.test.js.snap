// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Component 1`] = `null`;

exports[`Message Component 2`] = `
.c0 {
  width: 100%;
  margin-top: -1em;
  text-align: center;
}

.c0 p {
  display: inline-block;
  margin: 0;
  padding: 1em;
  background: #333;
  color: #fafafa;
  border-radius: 2px;
  overflow: hidden;
}

<div
  className="c0"
>
  <p>
    this is Message Component Text
  </p>
</div>
`;
